docs/developing/deploy_facilities/interacting_with_spl_tokens.md

```diff
@@ -4,100 +4,113 @@ proofedDate: na
 includedInSite: true
 approvedBy: na
-comments: #1 TODO update text #fix broken link -- see inline comment
+comments:
+boilerPlatable: TODO we could have an item providing demo
 ---
 
-The ERC-20 SPL wrapper contract provides access to native Solana tokens registered in the SPL token contract, through the ERC-20 interface.
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
-This allows Solana applications to interact with EVM (Solidity, Vyper, etc.) bytecode contracts. The ERC-20 SPL wrapper can also be used to transfer funds in Solana tokens using Ethereum wallets such as MetaMask.
+This page describes the ERC-20 SPL wrapper contract: the ERC-20 Factory Contract. This contract provides access to native Solana tokens, i.e. those registered in the SPL token contract, through the [ERC-20 standard interface](https://eips.ethereum.org/EIPS/eip-20). 
 
-The contract is implemented in Rust as part of the Neon EVM program.
+This allows Solana liquidity to be available to EVM (Solidity, Vyper, etc.) bytecode contracts, i.e. this ERC-20 SPL wrapper allows Ethereum wallets such as MetaMask to transact with SPL tokens. 
 
-Source code:
-  * [Rust source code](https://github.com/neonlabsorg/neon-evm/blob/c43345d7abf7af14aa840e6b15c0fc64b084bb2c/evm_loader/program/src/precompile_contracts.rs#L106)
-  * [Solidity wrapper source code](https://github.com/neonlabsorg/neon-evm/blob/6ac1734658f0fdcac09092bd98979d4f6fe4530d/evm_loader/solidity/erc20_for_spl.sol#LL12C11-L12C11)
+:::info
+To be able to use an SPL token from a Solana account balance, it must be transferred to a Neon EVM account via the [NeonPass SDK](/docs/developing/integrate/neon_transfer_sdk).
+:::
 
-### Contract Interface
+## ERC-20 Factory Contract
 
-```solidity
-interface IERC20 {
-    function decimals() external view returns (uint8);
-    function totalSupply() external view returns (uint256);
-    function balanceOf(address who) external view returns (uint256);
-    function allowance(address ow ERC-20 SPL-Wrapperer, address spender) external view returns (uint256);
-    function transfer(address to, uint256 value) external returns (bool);
-    function approve(address spender, uint256 value) external returns (bool);
-    function transferFrom(address from, address to, uint256 value) external returns (bool);
+The [ERC-20-for-SPL Factory Contract](https://github.com/neonlabsorg/neon-evm/blob/4bcae0f476721e5396916c43396ec85e465f878f/evm_loader/solidity/erc20_for_spl_factory.sol) provides a method to access a list of deployed contracts on the Neon EVM and to issue and register a new ERC-20-for-SPL contract. Once registered, these contracts are then deployed to Neon EVM and are available on the system-wide registry.
 
-    event Transfer(address indexed from, address indexed to, uint256 value);
-    event Approval(address indexed owner, address indexed spender, uint256 value);
+:::info
+Two addresses of the deployed contract will registered: Devnet and Mainnet.
+::: 
```

Comment:
The ERC20ForSpl factory contract is a system-wide registry which provides a method to access a list of...

This contract has the next addresses on clusters:
- Devnet: 0xF6b17787154C418d5773Ea22Afc87A95CAA3e957
- Mainnet: 0x6B226a13F5FE3A5cC488084C08bB905533804720